### ISSUE TITLE
Errors highlighted when using pipetools module

### ISSUE THREAD

**Issue by leldershaw** [microsoft/pylance-release#1886]
<h3> Environment data </h3>

-   Language Server version: 2021.9.3
-   OS and version: darwin x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.9.2
-   python.analysis.indexing: undefined
-   python.analysis.typeCheckingMode: off

<h3> Expected behaviour </h3>

Use of the symbol `|` to pipe tasks together using the `pipe` function in the `pipetools` package causing no highlighted Pylance errors.

<h3> Actual behaviour </h3>

Pylance is highlighting an an error with the use of `|` symbol before a `list` call when piping tasks together using `pipetools`. It states that `Alternative syntax for unions requires Python 3.10 or newer`. The code runs with no issues in Python 3.9.2 as `pipetools` is imported. This error goes away if Python 3.10 is used, but should not occur for v3.9.2. 

<h3> Logs </h3>

Python Language Server Log
```
No log output to provide
```

<h3> Code Snippet / Additional information </h3>

```bash
pip install pipetools
```

```python
from pipetools import pipe, where

odd_sum = (pipe | 
            range | 
            where(lambda x: x % 2) |    #This pipe symbol shows an error as it is before a `list` call
            list |
            sum)

print(odd_sum (5))
```

**Comment by erictraut** [comment]
Thanks for the bug report. I agree this is a false positive error. It will be fixed in the next release.

**Comment by jakebailey** [comment]
This issue has been fixed in version 2021.9.4, which we've just released. You can find the changelog here: https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#202194-29-september-2021
